File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string, std::vector<std::string>> str_map;

        int i;
        for (i = 0; i < strs.size(); i++) {
            std::string curr_str = strs[i];
            std::sort(curr_str.begin(), curr_str.end());

            str_map[curr_str].push_back(strs[i]);
        }

        std::vector<std::vector<std::string>> result;
        for (const auto &pair: str_map) {
            result.push_back(pair.second);
        }
        return result;
    }
};

// act, cat
//
// sort(act) --> act
// sort(cat) --> act

File: Data Structures & Algorithms/single-number/submission-1.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0

        for i in nums:
            result = result ^ i
        return result
        

# [3, 2, 3]
#

# 0011 
# 0011 XOR
# 0000

# 0000
# 0010 XOR
# 0010